_posts/2019-06-19-load-image-data.md

```diff
@@ -0,0 +1,1095 @@
+
+Once acquired this data presents a number of challenges. Algorithms often
+designed and tested to work on very small pieces of this data need to be scaled
+up to work on the full dataset. It might not be clear at the outset what will
+actually work and so exploration still plays a very big part of the whole
+process.
+
+Historically this analytical process has involved a lot of custom code. Often
+the analytical process is stitched together by a series of scripts possibly in
+several different languages that write various intermediate results to disk.
+Thanks to advances in modern tooling these process can be significantly
+improved. In this series of blogposts, we will outline ways for image
+scientists to leverage different tools to move towards a high level, friendly,
+cohesive, interactive analytical pipeline.
+
+Post Overview
+----------
+
+This post in particular focuses on loading and managing large stacks of image
+data in parallel from Python.
+
+Loading large image data can be a complex and often unique problem. Different
+groups may choose to store this across many files on disk, a commodity or
+custom database solution, or they may opt to store it in the cloud. Not all
+datasets within the same group may be treated the same for a variety of
+reasons. In short, this means loading data is a hard and expensive problem.
+
+Despite data being stored in many different ways, often groups want to reapply
+the same analytical pipeline to these datasets. However if the data pipeline is
+tightly coupled to a particular way of loading the data for later analytical
+steps, it may be very difficult if not impossible to reuse an existing
+pipeline. In other words, there is friction between the loading and analysis
+steps, which frustrates efforts to make things reusable.
+
+Having a modular and general way to load data makes it easy to present data
+stored differently in a standard way. Further having a standard way to present
+data to analytical pipelines allows that part of the pipeline to focus on what
+it does best, analysis! In general, this should decouple these to components in
+a way that improves the experience of users involved in all parts of the
+pipeline.
+
+
+Loading image data with Dask
+----------------------------
+
+Let's start again with our image data from the top of the post:
+
+```
+$ $ ls raw/ | head
+ex6-2_CamA_ch1_CAM1_stack0000_560nm_0000000msec_0001291795msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0001_560nm_0043748msec_0001335543msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0002_560nm_0087497msec_0001379292msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0003_560nm_0131245msec_0001423040msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0004_560nm_0174993msec_0001466788msecAbs_000x_000y_000z_0000t.tif
+```
+
+### Load a single sample image with Scikit-Image
+
+To load a single image, we use [Scikit-Image](https://scikit-image.org/):
+
+```python
+>>> import glob
```

Comment:
import os
os.chdir("path")

Reply:
I've changed the glob command from `glob("raw/*.tif")` to `glob("/path/to/raw/*.tif")` to hopefully avoid the confusion that I think you're concerned about.